src/entry_node.cpp

```diff
@@ -115,4 +117,218 @@ Status EntryNode::createShardedBlob(InferenceEngine::Blob::Ptr& dividedBlob, con
     return StatusCode::OK;
 }
 
+const Status EntryNode::validateNumberOfInputs(const tensorflow::serving::PredictRequest* request, const size_t expectedNumberOfInputs) {
+    if (request->inputs_size() < 0 || expectedNumberOfInputs != static_cast<size_t>(request->inputs_size())) {
+        std::stringstream ss;
+        ss << "Expected: " << expectedNumberOfInputs << "; Actual: " << request->inputs_size();
+        const std::string details = ss.str();
+        SPDLOG_DEBUG("Invalid number of inputs - {}", details);
+        return Status(StatusCode::INVALID_NO_OF_INPUTS, details);
+    }
+    return StatusCode::OK;
+}
+
+const Status EntryNode::checkIfShapeValuesNegative(const tensorflow::TensorProto& requestInput) {
+    for (int i = 0; i < requestInput.tensor_shape().dim_size(); i++) {
+        if (requestInput.tensor_shape().dim(i).size() < 0) {
+            const std::string details = "Negative dimension size is not acceptable: " + TensorInfo::tensorShapeToString(requestInput.tensor_shape());
+            SPDLOG_LOGGER_DEBUG(modelmanager_logger, "Invalid shape - {}", details);
```

Comment:
This seem to be bug copied from modelinstance.cpp -> we shouldn't use modelmanager_logger here

Reply:
fixed both